src/components/AvatarGroup/AvatarGroupCounter.tsx

```diff
@@ -1,25 +1,40 @@
-import React, { useCallback, useRef } from "react";
-import PropTypes from "prop-types";
+import React, { ReactElement, useCallback, useRef } from "react";
 import cx from "classnames";
-import Tooltip from "../Tooltip/Tooltip";
-import Avatar from "../Avatar/Avatar";
+import { AvatarProps } from "../Avatar/Avatar";
 import Counter from "../Counter/Counter";
 import MenuButton from "../MenuButton/MenuButton";
 import Menu from "../Menu/Menu/Menu";
 import AvatarMenuItem from "../Menu/MenuItem/AvatarMenuItem";
 import AvatarGroupCounterTooltipContainer from "./AvatarGroupCounterTooltipContainer";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import { AvatarSizes, AvatarTypes } from "../Avatar/AvatarConstants";
+import { getStyle } from "../../helpers/typesciptCssModulesHelper";
+import { AvatarGroupCounterVisualProps } from "./AvatarGroup";
+import { avatarOnClick } from "./AvatarGroupHelper";
 import styles from "./AvatarGroupCounter.module.scss";
 
-const avatarOnClick = avatarProps => {
-  return avatarProps?.onClick && (() => avatarProps.onClick(avatarProps?.id));
-};
+interface AvatarGroupCounterProps extends VibeComponentProps {
+  /**
+   * Array of Avatar elements
+   */
+  counterTooltipAvatars?: ReactElement<AvatarProps>[];
+  /**
+   * AvatarGroupCounterVisualProps: props for counter
+   */
+  counterProps?: AvatarGroupCounterVisualProps;
+  // TODO ts-migration replace with TooltipProps when Tooltip is converted to TS
+  counterTooltipCustomProps?: any;
+  counterTooltipIsVirtualizedList?: boolean;
+  size?: AvatarSizes;
+  type?: AvatarTypes;
+}
 
-const AvatarGroupCounter = ({
-  counterTooltipAvatars,
+const AvatarGroupCounter: React.FC<AvatarGroupCounterProps> = ({
+  counterTooltipAvatars = [],
   counterProps,
   counterTooltipCustomProps,
-  counterTooltipIsVirtualizedList,
-  size,
+  counterTooltipIsVirtualizedList = false,
+  size = AvatarSizes.MEDIUM,
```

Comment:
When we implement enums we use names in the singular and not the plural
AvatarSizes => AvatarSize